### ISSUE TITLE
Hover county ID sometimes incorrect, or NA

### ISSUE THREAD

**Issue by grrmartin-USGS** [USGS-R/wateRuse#136]
Some county identifiers incorrect or NA in Indiana and Kentucky. Getting NA for Marion County (Indianapolis, largest population) in center of state.  Should be FIPS 097.  Most northwestern county reported as 083, Knox. (Should be 089, Lake County). Most northeastern county reported as 039, Elkhart.  (Should be 151, Steuben County) Some counties in Kentucky correct, but others incorrect: eastern most county reported as 017 (Bourbon), Should be 195 (Pike).  Display of county identifier sometimes fades out, as though it is looking up data, then returns, sometimes changed to NA.

**Comment by jshourds-usgs** [comment]
@grrmartin-USGS -- zoiks!  sounds poltergiesty!  Will check on NJ at some point this week...  I do remember having trouble with the NJ's HUCs and county pops using one of the files but can't remember which file at this point.  Do you remember what file you added in?

**Comment by ldecicco-USGS** [comment]
The hover for the maps in general could probably be done better...but I'm not sure how without investing some time exploring that.

**Comment by dblodgett-usgs** [comment]
I'm having trouble with hover on the Compare Two Years plot... I'm using the 2010 compilation data as my source and looking at the whole state of CA. It works on the Compare Two Elements and choropleth, but not the compare two years. Anyone else seeing this?

**Comment by rwdudley-usgs** [comment]
I tested it this morning using the data sets below and didn't see the issue. Could you point me to the 2010 compilation data? Not sure where that is. 

exampleAWUDSdump.txt, wu_countydata_20160415.txt (for Alaska), Export_2005_County.xlsx, Export_2010_County.xlsx, Export_2015_County.xlsx, Export_2016_HUC-8.xlsx, State compiler data (county) for Maine 1985-2010, and State compiler data (HUC8) for Maine 1985-1995

**Comment by dblodgett-usgs** [comment]
Not sure where I got it, it's been a while. I just uploaded it to the 'INTERNAL ONLY example' google drive folder.

**Comment by rwdudley-usgs** [comment]
Thanks. I downloaded it and tested AK - fine. Tested CA, and I can see what you mean. Does the tolerance for hover change with the number of points being displayed? Because it seems like I can get the hover text to display when many points (all CA counties) are on the plot, but the cursor has to hit a very specific pixel. But if I change the display to show only a half-dozen counties, the hover text appears as long as I'm somewhat close.

**Comment by dblodgett-usgs** [comment]
Seem like this is an issue with the basic hover functionality we are using. It needs to be reworked completely to solve this issue. If this becomes a major priority, we will revisit but that will require a different mapping approach.

**Comment by ldecicco-USGS** [comment]
If you guys would pull in the latest changes, rebuild, then let me know if the hover seems more accurate.

**Comment by ldecicco-USGS** [comment]
@grrmartin-USGS see if using plotly has solved this issue. If so, close it up.

**Comment by grrmartin-USGS** [comment]
It is displaying a pop-up rectangle for each county nicely--however, there is nothing printed in the rectangles (they are just solid colors--same color as the county). 

Also IN and other plots are distorted now. The scaling is off from a normal projection, which would enlarge about 200% vertically:     (Unless this is intended as a perspective view.)
![choroplethdistortedincapture](https://cloud.githubusercontent.com/assets/18426661/19421175/7edb2082-93c8-11e6-8d2d-8296a3eb85a3.JPG)

**Comment by rwdudley-usgs** [comment]
Yes I've been seeing the same effect for all maps - the map just automatically fills the available plot space with no regard for proper spatial scale. It's fine for the time being, but we'll probably want to figure out a way to enforce the right scale at some point.

**Comment by ldecicco-USGS** [comment]
I actually think I improved the scaling a bit in a recent pull request. If you can do an upstream pull request, that should help.

**Comment by rwdudley-usgs** [comment]
Yes, much improved! Thanks!

**Comment by grrmartin-USGS** [comment]
I'm still not seeing the county name, etc in the boxes that pop up when
hovering. The boxes are blank, the same shade as the county.

Gary Martin (grmartin@usgs.gov)

Hydrologist

U.S. Geological Survey

Kentucky Water Science Center

9818 Bluegrass Parkway

Louisville, Kentucky

40299-1906

(502) 609-1383 (cell)

(502) 493-1914 (phone)

(502) 493-1909 (fax)

**Comment by ldecicco-USGS** [comment]
Can you describe what you are doing? ie...what data set, what state, what browser? I haven't been able to reproduce that.

**Comment by grrmartin-USGS** [comment]
That must be it--the data set is from the sprint:

\INTERNAL_ONLY_example_data_files-2016-05-20\INTERNAL ONLY example data
files\State compiler dump with state data set for one
year\wu_countydata_20160415.txt.

Running in IE11.  The hover boxes are blank in all states I checked, IN KY,
AL.

I tried my AWUDS export excel files, but still blank in IE11.

It works, the hover boxes display,  with both files types, in _Chrome._

Gary Martin (grmartin@usgs.gov)

Hydrologist

U.S. Geological Survey

Kentucky Water Science Center

9818 Bluegrass Parkway

Louisville, Kentucky

40299-1906

(502) 609-1383 (cell)

(502) 493-1914 (phone)

(502) 493-1909 (fax)
